docs/src/content/config.ts

```diff
@@ -0,0 +1,28 @@
+import { defineCollection, z } from "astro:content";
+
+export const collections = {
+  docs: defineCollection({
+    type: "content",
+    schema: z.object({
+      /**
+       * The title would be used in the navbar.
+       */
+      title: z.string().min(1),
+
+      /**
+       * The category for the content. This would be used in the navbar to group
+       * the content together.
+       */
+      category: z.string().min(1),
+
+      /**
+       * The priority defines the sorting order *within* the category.
+       */
+      priority: z.number().positive().int().optional(),
+
+      isV1: z.literal(true as const).optional(),
```

Comment:
**nit:** maybe this is premature, but i'd rather this be a literal of versions rather than a boolean

Reply:
I'm thinking of doing this via the folder (`v1/`) instead of having any sort of version flag for the same reason that I've mentioned in other places: I don't want us to "maintain" a v1 site, hopefully, what we're doing now is the last we'll touch anything related to v1. I prefer not to build the whole site in a way that signals multi-version support.

My goal with this abstraction is to add extra documentation articles to the docs so that we remove the explanations from the index page so we can make it splashier and less verbose.